types/dom-chromium-ai/index.d.ts

```diff
@@ -0,0 +1,253 @@
+}
+
+interface AI {
+    readonly assistant: AIAssistantFactory;
+    readonly summarizer: AISummarizerFactory;
+    readonly writer: AIWriterFactory;
+    readonly rewriter: AIRewriterFactory;
+}
+
+interface AICreateMonitor {
+    ondownloadprogress: (event: DownloadProgressEvent) => void;
+
+    addEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: AddEventListenerOptions | boolean,
+    ): void;
+    removeEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: EventListenerOptions | boolean,
+    ): void;
+}
+
+interface DownloadProgressEvent extends Event {
+    readonly loaded: number;
+    readonly total: number;
+}
+
+type AICreateMonitorCallback = (monitor: AICreateMonitor) => void;
+
+type AICapabilityAvailability = "readily" | "after-download" | "no";
+
+interface AIReadableStream extends ReadableStream<string> {
+    [Symbol.asyncIterator](): AsyncIterableIterator<string>;
+}
+
+// Assistant
+// https://github.com/explainers-by-googlers/prompt-api/#full-api-surface-in-web-idl
+
+interface AIAssistantFactory {
+    create(options?: AIAssistantCreateOptions): Promise<AIAssistant>;
+    capabilities(): Promise<AIAssistantCapabilities>;
+}
+
+interface AIAssistantCreateOptions {
+    signal?: AbortSignal;
+    monitor?: AICreateMonitorCallback;
+
+    systemPrompt?: string;
+    initialPrompts?: AIAssistantPrompt[];
+    topK?: number;
+    temperature?: number;
+}
+
+type AIAssistantPromptRole = "system" | "user" | "assistant";
+
+interface AIAssistantPrompt {
+    role?: AIAssistantPromptRole;
+    content?: string;
+}
+
+interface AIAssistant {
+    prompt(input: string, options?: AIAssistantPromptOptions): Promise<string>;
+    promptStreaming(input: string, options?: AIAssistantPromptOptions): AIReadableStream;
+
+    countPromptTokens(input: string, options?: AIAssistantPromptOptions): Promise<number>;
+    readonly maxTokens: number;
+    readonly tokensSoFar: number;
+    readonly tokensLeft: number;
+
+    readonly topK: number;
+    readonly temperature: number;
+
+    oncontextoverflow: (event: Event) => void;
+
+    addEventListener(
+        type: "contextoverflow",
+        listener: (event: Event) => void,
+        options?: AddEventListenerOptions | boolean,
+    ): void;
+    removeEventListener(
+        type: "contextoverflow",
+        listener: (event: Event) => void,
+        options?: EventListenerOptions | boolean,
+    ): void;
+
+    clone(): Promise<AIAssistant>;
+    destroy(): void;
+}
+
+interface AIAssistantPromptOptions {
+    signal?: AbortSignal;
+}
+
+interface AIAssistantCapabilities {
+    readonly available: AICapabilityAvailability;
+
+    readonly defaultTopK: number | null;
+    readonly maxTopK: number | null;
+    readonly defaultTemperature: number | null;
+
+    supportsLanguage(languageTag: string): AICapabilityAvailability;
```

Comment:
There might be a type for language tags like `"de"` somewhere, because `Intl.*` uses this.

Reply:
Done! I'm using [`Intl.UnicodeBCP47LocaleIdentifier`](https://github.com/microsoft/TypeScript/blob/a709f9899c2a544b6de65a0f2623ecbbe1394eab/src/lib/es2020.intl.d.ts#L10) (which however is simply an alias for `string`).